docs/v3tov4.md

```diff
@@ -0,0 +1,283 @@
+search and filter them.
+
+To add a new Dimensions just call on a metric 
+
+`func (i *<Gauge|Count|Summary>) AddDimension(key string, value interface{})`
+
+If the `key` already exists, then the previous value will be overwritten with the new one.
+
+#### Adding metrics to an Entity
+
+Once you have a metric, it can be attached to an Entity. The metric will belong to that Entity and it will be decorated
+with the Entity's tags.
+
+To add a metric to an entity just call the method on an Entity
+
+`func (e *Entity) AddMetric(metric)`
+
+#### Adding metrics to an Integration
+
+Metrics can also be added to an Integration. Those metrics won't be attached to any Entity.
+
+To add a metric to an Integrations just call the method inside the `integration` namespace
+
+`func (e *Integration) AddMetric(metric)`
+
+#### Metrics summary
+
+| type       | v3 | v4 | mapping                                                                          |
+|------------|----|----|---------------------------------------------------------------------------------|
+| gauge      | ✅  | ✅  | integration.Gauge(timestamp, name, value)                                     |
+| rate/prate | ✅  | ❌  | use a gauge and the NRQL rate function                                        |
+| delta      | ✅  | ❌  | use a gauge and calculate the delta                                           |
+| pdelta     | ✅  | ✅  | integration.PDelta(timestamp, name, value)                                    |
+| attribute  | ✅  | ❌  | add dimensions to the metrics                                                 |
+| count      | ❌  | ✅  | integration.Count(timestamp, interval, name, count)                           |
+| summary    | ❌  | ✅  | integration.Summary(timestamp, interval, name, count, average, sum, min, max) |
+
+### Events
+
+The API for creating events has changed a little bit. Events can be attached to an Entity or Integration.
+
+In SDK v3, events were created like this
+
+`func New(summary, category string) *Event`
+
+Now, for creating an event, you need to call this method inside the `integration` package
+
+`func NewEvent(timestamp time.Time, summary, category string) *Event`
+
+The parameters are the same as before with the exception of the `timestamp`. `summary` cannot be empty.
+ 
+#### Adding attributes to an Event
+
+In SDK v3, attributes were added to events using this method
+
+`func AddCustomAttributes(e *Event, customAttributes []attribute.Attribute)`
+
+In SDK v4, you need to use this other method for adding a single attribute
+
+`func (e *Event) AddAttribute(key string, value interface{})`
+
+If the `key` already exists, then the previous value will be overwritten with the new one.
+
+#### Adding an event to an Entity
+
+In both SDKs, the same method exists for adding an event to an entity
+
+`func (e *Entity) AddEvent(event *integration.Event) error`
+
+The event will be decorated with the Entity's tags.
+
+#### Adding an event to an Integration
```

Comment:
I think this is misleading, because it will be attached to the HOST entity.

Reply:
i guess we could have a single entity called "host" to store this